### ISSUE TITLE
token not receiving

### ISSUE THREAD

**Issue by nullhandler** [espresso3389/oauth2_custom_uri_scheme#4]
The token is redirected to the redirect URL. Still, the plugin doesnt retrieve the token and close the custom tab. The redirect URL I have used is ```http://in.selvasoft.lite123```
Also this warning while compiling the app
```
warning: [deprecation] getFlutterEngine() in FlutterPluginBinding has been deprecated
setupChannel(binding.getFlutterEngine().getDartExecutor(), binding.getApplicationContext());
```

**Comment by espresso3389** [comment]
Android or iOS?
For Android, you should setup your redirect URL on AndroidManifest.xml as explained [here](https://github.com/espresso3389/oauth2_custom_uri_scheme#additional-settings-on-android).

**Comment by nullhandler** [comment]
for android. Yeah I followed those steps. Still its not catching.

**Comment by selvasoft** [comment]
thanks. that problem is solved.

**Comment by nullhandler** [comment]
the chrome custom tabs is closing. but the token object is null. Try running your example.
See the following in the logs
```
I/flutter (  427): Custom Chrome Tab seems to be closed.
I/flutter (  427): PlatformException(error, The new embedding does not support the old FlutterView., null)
```

**Comment by espresso3389** [comment]
Could you please show me your AndroidManifest.xml?

**Comment by nullhandler** [comment]
[here](https://pastebin.com/raw/iZbiLSCK) you go. I am just running the code from your example folder. Are you in telegram? Can you try running your example code?

**Comment by nullhandler** [comment]
@espresso3389

**Comment by espresso3389** [comment]
You could not use `http://in.selvasoft.lite123` as redirect URL unless you properly configure deep-link; futhermore, http is not good for our purpose anyway.

To make redirect URL work, we should use URL like `myapp://foobar` or such.

In your [AndroidManifest.xml](https://pastebin.com/raw/iZbiLSCK), you have the following redirect URL config:

```xml
<data android:scheme="com.example.redirect43763246328" android:host="callback" />
```

And the redirect URL must be `com.example.redirect43763246328:callback`.

**Comment by nullhandler** [comment]
@espresso3389 I am just running your example. Where did you found `http://in.selvasoft.lite123`? Try running your example folder.

**Comment by espresso3389** [comment]
For what service do you run the sample? Your redirect URI must be registered on the service website.

**Comment by nullhandler** [comment]
@espresso3389 I registered at the reddit. Is there any better way to contact you?

**Comment by mprabhu33** [comment]
Am getting same error. @selvasoft Are you able to resolve the issue?

**Comment by nullhandler** [comment]
@mprabhu33 No its still the same. So, I implemented on my own.

**Comment by espresso3389** [comment]
> the chrome custom tabs is closing. but the token object is null. Try running your example.
> See the following in the logs
> 
> ```
> I/flutter (  427): Custom Chrome Tab seems to be closed.
> I/flutter (  427): PlatformException(error, The new embedding does not support the old FlutterView., null)
> ```

Wow, I realized the issue is identical to #5 just now.
And, at least, on my environmens, it worked without any issues. On #5, I just modified the code without knowing the actual cause of the issue but just change the way to obtain Context according to the [ShimRegistrar.java code](https://github.com/flutter/engine/blob/master/shell/platform/android/io/flutter/embedding/engine/plugins/shim/ShimRegistrar.java)

So anyway, the change was already commited. Please reference #5 for that.

**Comment by espresso3389** [comment]
Sorry, the previous commit does not fix the issue and 7d4d869ea8e0da140aa75ae7fa39367a0cf2418e (0.3.8) finally fixes the issue.

**Comment by nullhandler** [comment]
thanks for the library @espresso3389
